server-core/engine/src/main/java/io/infectnet/server/engine/core/world/strategy/AStarPathFinderStrategy.java

```diff
@@ -0,0 +1,298 @@
+        }
+        /* Even if no path found to the target Position,
+         the path that has already been constructed in the direction of the target will be returned. */
+
+        return getPathFromNodes(world, start, current);
+
+    }
+
+    /**
+     * Calculating the cost of a Tile, could be used for more diversive Tile types,
+     * in this case with only one type to move on a constant function is  enough.
+     * @param neighbour the node representing  of the Tile given
+     * @return the cost of stepping to the given Position
+     */
+    private int calculateCostOfTile(Node neighbour) {
+        Tile tile = world.getTileByPosition(neighbour.getPosition());
+        switch (tile.getType()){
+            case CAVE: return 1;
+            case ROCK: return 100;
+            default: return 0;
+        }
+    }
+
+    /**
+     * Constructs the Path, which can be returned as the result of the A* algorithm.
+     * Usis recursive path building, going from the last Position in the Path, bindig
+     * its parent Node to it, and so on. Until it reaches the starting Position.
+     * @param world the map where we search
+     * @param start the starting Position
+     * @param current the end Position of the current Path, may not be the original target
+     * @return a linked list of the path containing the target at the front and the start at the back
+     */
+    private LinkedList<Tile> getPathFromNodes(World world, Position start, Node current) {
+        LinkedList<Tile> path = new LinkedList<>();
+        Node node = current;
+
+        while (node.parent != null) {
+            path.add(world.getTileByPosition(node.getPosition()));
+            node = node.parent;
+        }
+        path.add(world.getTileByPosition(start));
+        return path;
+    }
+
+    /**
+     * Finds the next Node to expand, which have to have the lowest cost.
+     * @return the chosen Node
+     */
+    private Node findNodeToExpand() {
+        Node current = null;
+
+        for (Node node : openNodes) {
+            if (current == null || node.getSumCost() < current.getSumCost()) {
+                current = node;
+            }
+        }
+
+        return current;
+    }
+
+    /**
+     * Always repositions the given target Position to one, that is not occupied or not a {@code ROCK}.
+     * @return the new position nearby the original target
+     */
+    private Position resetTarget() {
+        Position newTarget =  targetPos;
+        Node targetNode = new Node(targetPos);
+
+        for(Node node : targetNode.getNeighbours()){
+            if(world.isPositionValidTile(node.getPosition())
+                    && !world.getTileByPosition(node.getPosition()).isBlockedOrOccupied()){
+                newTarget = node.getPosition();
+                break;
+            }
+        }
+        return newTarget;
+    }
+
+    /**
+     * Inner class to represent the special cell required by the A* algorithm
+     */
+    private class Node {
```

Comment:
Please add the `static` modifier to this class. It does not access its containing instance so a static nested class is perfectly reasonable.